src/main/java/io/pravega/connectors/hadoop/PravegaOutputRecordWriter.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.pravega.connectors.hadoop;
+
+import io.pravega.client.stream.EventStreamWriter;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.UUID;
+
+/**
+ * A RecordWriter that can write events to Pravega.
+ */
+@NotThreadSafe
+public class PravegaOutputRecordWriter<V> extends RecordWriter<NullWritable, V> {
+
+    private static final Logger log = LoggerFactory.getLogger(PravegaOutputRecordWriter.class);
+
+    protected final AtomicInteger pendingWritesCount;
+    private final AtomicReference<Throwable> writeError;
+    private final ExecutorService executorService;
+    private final EventStreamWriter writer;
+    private final PravegaEventRouter<V> pravegaEventRouter;
+
+    public PravegaOutputRecordWriter(EventStreamWriter writer, PravegaEventRouter<V> pravegaEventRouter) {
+        this.writer = writer;
+        this.pendingWritesCount = new AtomicInteger(0);
+        this.writeError = new AtomicReference<>(null);
+        this.executorService = Executors.newFixedThreadPool(5);
```

Comment:
Is number 5 based on some experience? Please consider make it configurable, or at least a static variable if hard-coding is necessary.

Reply:
Moved the hard coded value to a variable